src/Workspaces/Core/Portable/SolutionCrawler/IIncrementalAnalyzerExtensions.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.ExternalAccess.UnitTesting;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.SolutionCrawler
+{
+    internal static partial class IIncrementalAnalyzerExtensions
+    {
+        public static BackgroundAnalysisScope GetOverriddenBackgroundAnalysisScope(this IIncrementalAnalyzer incrementalAnalyzer, OptionSet options, BackgroundAnalysisScope defaultBackgroundAnalysisScope)
+        {
+            // Unit testing analyzer has special semantics for analysis scope.
+            if (incrementalAnalyzer is UnitTestingIncrementalAnalyzer unitTestingAnalyzer)
+            {
+                return unitTestingAnalyzer.GetBackgroundAnalysisScope(options);
+            }
+
+            // TODO: Remove the below if statement once SourceBasedTestDiscoveryIncrementalAnalyzer has been switched to UnitTestingIncrementalAnalyzer
+            if (incrementalAnalyzer.GetType().FullName == "Microsoft.CodeAnalysis.UnitTesting.SourceBasedTestDiscovery.SourceBasedTestDiscoveryIncrementalAnalyzer")
+            {
+                return BackgroundAnalysisScope.FullSolution;
+            }
+
+            return defaultBackgroundAnalysisScope;
```

Comment:
shouldn't we be checking the LowVM mode somewhere?  if it's triggered, we would return teh ActiveFile/Minimal scope right?

Reply:
Hmm potentially we can avoid calling this API when LowVM mode is enabled. I will add that check.

Reply:
Actually, I now recollect that I had initially done that. However, this would mean Test explorer functionality would be switched to active file only in this scenario. This would be a behavior change for Test explorer's background analysis, which they do not desire as part of this PR.